src/leap/keymanager/__init__.py

```diff
@@ -185,34 +190,46 @@ def _key_class_from_type(self, ktype):
             lambda klass: klass.__name__ == ktype,
             self._wrapper_map).pop()
 
-    def _get(self, uri, data=None):
+    @defer.inlineCallbacks
+    def _get_json(self, address):
         """
         Send a GET request to C{uri} containing C{data}.
 
         :param uri: The URI of the request.
         :type uri: str
-        :param data: The body of the request.
-        :type data: dict, str or file
 
-        :return: The response to the request.
-        :rtype: requests.Response
+        :return: A deferred that will be fired with the GET response
+        :rtype: Deferred
         """
         leap_assert(
             self._ca_cert_path is not None,
             'We need the CA certificate path!')
```

Comment:
Actually, I don't like asserts that much, but in this case it will ensure that we aren't doing this without providing a specific cert instead of system bundle. To me this will help avoiding to make the request without the proper cert pinning and bonus point for a longer message.
Maybe the assert could verify more stuff, but I don't feel comfortable removing it.
Since I am new to keymanager and that can be an invalid concern, that's up to you. :)